File: .skills/openclaw-skills/skills/gatex402/x402-agentic-creation/src/lib/api-client.ts
export class GateX402Client {
    private baseUrl = 'https://api.gatex402.dev';

    constructor(private managementToken?: string) { }

    private get headers() {
        const headers: Record<string, string> = {
            'Content-Type': 'application/json',
        };
        if (this.managementToken) {
            headers['Authorization'] = `Bearer ${this.managementToken}`;
        }
        return headers;
    }

    async post(path: string, body: any) {
        const res = await fetch(`${this.baseUrl}${path}`, {
            method: 'POST',
            headers: this.headers,
            body: JSON.stringify(body),
        });
        if (!res.ok) {
            const error = await res.json().catch(() => ({ message: 'Unknown error' }));
            const detail = typeof error.details === 'string' ? error.details : (error.errors ? JSON.stringify(error.errors) : '');
            const msg = [error.message, detail].filter(Boolean).join(' ');
            throw new Error(msg || `API Error: ${res.status}`);
        }
        return res.json();
    }

    async get(path: string) {
        const res = await fetch(`${this.baseUrl}${path}`, {
            headers: this.headers,
        });
        if (!res.ok) {
            const error = await res.json().catch(() => ({ message: 'Unknown error' }));
            const detail = typeof error.details === 'string' ? error.details : (error.errors ? JSON.stringify(error.errors) : '');
            const msg = [error.message, detail].filter(Boolean).join(' ');
            throw new Error(msg || `API Error: ${res.status}`);
        }
        return res.json();
    }

    async patch(path: string, body: any) {
        const res = await fetch(`${this.baseUrl}${path}`, {
            method: 'PATCH',
            headers: this.headers,
            body: JSON.stringify(body),
        });
        if (!res.ok) {
            const error = await res.json().catch(() => ({ message: 'Unknown error' }));
            const detail = typeof error.details === 'string' ? error.details : (error.errors ? JSON.stringify(error.errors) : '');
            const msg = [error.message, detail].filter(Boolean).join(' ');
            throw new Error(msg || `API Error: ${res.status}`);
        }
        return res.json();
    }
}

File: .skills/openclaw-skills/skills/1044197988/ht-skills/lib/api_client.py
# -*- coding: utf-8 -*-
"""
Skill API 客户端：带 Token 的 HTTP 请求封装
author: 灏天文库
"""

import os
import json
import sys
from pathlib import Path
from typing import Optional, Dict, Any

try:
    import requests
except ImportError:
    requests = None


def get_client_root() -> Path:
    """client 目录根路径"""
    return Path(__file__).resolve().parent.parent


def load_config() -> Dict[str, Any]:
    """加载 client config.json，支持环境变量覆盖"""
    root = get_client_root()
    config_path = root / "config.json"
    config = {}
    if config_path.exists():
        with open(config_path, "r", encoding="utf-8") as f:
            config = json.load(f)
    # 环境变量优先
    server_url = os.environ.get("HT_SKILL_SERVER_URL") or config.get("server_base_url")
    token = os.environ.get("HT_SKILL_TOKEN") or config.get("token")
    return {"server_base_url": server_url, "token": token}


def request(
    method: str,
    path: str,
    params: Optional[Dict] = None,
    json_body: Optional[Dict] = None,
) -> Dict[str, Any]:
    """
    发送请求到 Skill 服务端，自动附带 Token
    method: GET, POST, PATCH 等
    path: 如 /api/collections（不含 base_url）
    params: query 参数（GET）
    json_body: JSON 请求体（POST/PATCH）
    返回解析后的 JSON，失败时输出错误并 sys.exit(1)
    """
    if not requests:
        print(json.dumps({"success": False, "error": "请安装 requests: pip install requests"}, ensure_ascii=False, indent=2))
        sys.exit(1)
    cfg = load_config()
    base_url = (cfg.get("server_base_url") or "").rstrip("/")
    token = cfg.get("token")
    if not base_url:
        print(json.dumps({"success": False, "error": "未配置 server_base_url，请在 config.json 或环境变量 HT_SKILL_SERVER_URL 中设置"}, ensure_ascii=False, indent=2))
        sys.exit(1)
    if not token:
        print(json.dumps({"success": False, "error": "未配置 token，请在 config.json 或环境变量 HT_SKILL_TOKEN 中设置"}, ensure_ascii=False, indent=2))
        sys.exit(1)
    url = f"{base_url}{path}"
    headers = {
        "Authorization": f"Bearer {token}",
        "Content-Type": "application/json",
    }
    try:
        resp = requests.request(
            method=method,
            url=url,
            params=params,
            json=json_body,
            headers=headers,
            timeout=60,
        )
        data = resp.json() if resp.content else {}
        if resp.status_code >= 400:
            err = data.get("error") or data.get("detail") or resp.text or f"HTTP {resp.status_code}"
            print(json.dumps({"success": False, "error": err, "status_code": resp.status_code}, ensure_ascii=False, indent=2))
            sys.exit(1)
        return data
    except requests.exceptions.RequestException as e:
        print(json.dumps({"success": False, "error": str(e), "message": "请求服务端失败"}, ensure_ascii=False, indent=2))
        sys.exit(1)
    except json.JSONDecodeError as e:
        print(json.dumps({"success": False, "error": f"响应解析失败: {e}"}, ensure_ascii=False, indent=2))
        sys.exit(1)


def output_result(data: dict) -> None:
    """输出 JSON 结果到 stdout"""
    print(json.dumps(data, ensure_ascii=False, indent=2))
